src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -47,6 +50,12 @@ private void MakeInterpolatedStringFormat(BoundInterpolatedString node, out Boun
                 {
                     // this is one of the literal parts
                     formatString.Builder.Append(part.ConstantValue.StringValue);
+                    
+                    // no point in continuing to unescape when the result will be unused
+                    if (isSimpleConcatString) 
```

Comment:
only when `i == 0`; if you processed a part earlier that is a `BoundStringInsert` and has Alignment or FormatString properties, it will be false for all literals afterwords. For example `$"{0:c} dollars"` would be false when `i == 1`